apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -660,6 +660,18 @@ type RebootAnnotationArguments struct {
 	Mode RebootMode `json:"mode"`
 }
 
+type DetachedDeprovisionAction string
+
+const (
+	DetachedDeprovisionActionDelay  = "delay"
+	DetachedDeprovisionActionDelete = "delete"
+)
+
+type DetachedAnnotationArguments struct {
+	// DeprovisionAction indicates the desired deprovition logic when the detached annotation is present
+	DeprovisionAction DetachedDeprovisionAction `json:"deprovisionAction,omitempty"`
```

Comment:
'deprovisionAction' doesn't seem like the right term, because it doesn't control what happens when you deprovision, but rather what happens when you delete. In fact as long as this annotation is set, we never deprovision at all.